conans/client/conan_api.py

```diff
@@ -74,6 +74,13 @@ def wrapper(*args, **kwargs):
             with tools.environment_append(api._cache.config.env_vars):
                 # Patch the globals in tools
                 return f(*args, **kwargs)
+        except NotFoundException as exc:
```

Comment:
Hi @czoido 

I don't think this should be handled here, but much sooner. Here it is too late, and there are other things that could cause a NotFoundException (e.g. trying to download a .zip for the ``conan config install``).

It is the exception itself who should encode a better error message as close to the raise as possible, by the one that has the context (remote_name, ref, or whatever makes sense)